src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/client/HmppsAuthClient.kt

```diff
@@ -0,0 +1,27 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client
+
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.config.bearerToken
+
+/**
+ * Client class for interacting with `hmpps-auth`s REST API
+ */
+@Component
+class HmppsAuthClient(private val hmppsAuthWebClient: WebClient) {
+
+  /**
+   * Calls the `/user/me` endpoint with the given user token, and returns the user's active caseload ID from the response.
+   */
+  fun getUserActiveCaseLoadId(userToken: String): String? {
+    val userProfile = hmppsAuthWebClient
+      .get()
+      .uri("/api/user/me")
+      .bearerToken(userToken)
+      .retrieve()
+      .bodyToMono(Map::class.java)
+      .block()
+
+    return userProfile?.get("activeCaseLoadId")?.toString()
+  }
```

Comment:
I've not added any error handling to this at all yet - so if HMPPS Auth returns a non-200, all manner of badness is likely to happen! Not really tested it yet, but I guess we need to think about what that might mean for our user journey / how we want to handle it